src/ir_passes/mono/monomorphize.rs

```diff
@@ -88,19 +72,26 @@ impl<'ctx> Monomorphize<'ctx> {
 
         // If we've already processed this or queued this for processing, return the component
         if let Some(&name) = self.processed.get(&key) {
-            return (name, vec![]);
+            return name;
         }
 
-        if let Some((name, _)) = self.queue.get(&key) {
```

Comment:
Can we get a comment somewhere here, or in the PR description maybe about how the structure of the pass has changed, for me to understand better and also so we have it for the future? I understand why it had to be changed from #310, but maybe documenting the change somewhere would be helpful

Reply:
Good point! I can add something to the description of the monomorphize pass but I also think we should write a doc page on filamenthdl.com/docs that describes more internal details of the compiler so that it's easier to remember in the future.

Reply:
Done!